vulnerabilities/importers/debian.py

```diff
@@ -21,115 +21,194 @@
 
             for release_name, release_record in record["releases"].items():
-                if not release_record.get("repositories", {}).get(release_name):
-                    continue
+                version = get_item(release_record, "repositories", release_name)
 
-                version = release_record["repositories"][release_name]
+                if not version:
+                    logger.error(
+                        f"Version not found for {release_name} in {record} in package {pkg_name}"
+                    )
+                    continue
 
                 if version in ignored_versions:
+                    logger.error(f"Ignoring version {version} in {record} in package {pkg_name}")
                     continue
 
                 purl = PackageURL(
                     name=pkg_name,
                     type="deb",
                     namespace="debian",
-                    version=version,
                     qualifiers={"distro": release_name},
                 )
 
                 if release_record.get("status", "") == "resolved":
-                    resolved_purls.append(purl)
+                    fixed_versions.append(version)
                 else:
-                    impacted_purls.append(purl)
+                    affected_versions.append(version)
 
                 if (
                     "fixed_version" in release_record
                     and release_record["fixed_version"] not in ignored_versions
                 ):
-                    resolved_purls.append(
-                        PackageURL(
-                            name=pkg_name,
-                            type="deb",
-                            namespace="debian",
-                            version=release_record["fixed_version"],
-                            qualifiers={"distro": release_name},
-                        )
-                    )
+                    fixed_versions.append(release_record["fixed_version"])
 
             references = []
             debianbug = record.get("debianbug")
             if debianbug:
                 bug_url = f"https://bugs.debian.org/cgi-bin/bugreport.cgi?bug={debianbug}"
-                references.append(Reference(url=bug_url, reference_id=debianbug))
-            advisories.append(
-                AdvisoryData(
-                    vulnerability_id=cve_id,
-                    affected_packages=nearest_patched_package(impacted_purls, resolved_purls),
-                    summary=record.get("description", ""),
-                    references=references,
+                references.append(Reference(url=bug_url, reference_id=str(debianbug)))
+            affected_versions = list(dict.fromkeys(affected_versions))
+            fixed_versions = list(dict.fromkeys(fixed_versions))
+            affected_version_range = (
+                DebianVersionRange.from_versions(affected_versions) if affected_versions else None
```

Comment:
Here it may be easier to read:
```Python
if affected_versions:
    affected_version_range = DebianVersionRange.from_versions(affected_versions)
else:
    affected_version_range = None
```